packages/api/lib/writeRecords/write-granules.js

```diff
@@ -450,22 +453,30 @@ const _writeGranuleFiles = async ({
  * @param {Object} params.granule - An API granule object
  * @param {Object} params.provider - An API provider object
  *
-* @returns {Promise<Array>} - A list of file objects once resolved
+* @returns {Promise<Array> | undefined} - A list of file objects once resolved
  */
 const _generateFilesFromGranule = async ({
   granule,
   provider,
 }) => {
-  const { files = [] } = granule;
+  // FUTURE: null files are currently not supported in update payloads
+  // RDS Phase 3 should revise logic to accept an explict null value
+  if (granule.files === null) {
+    throw new CumulusMessageError('granule.files must not be null');
+  }
```

Comment:
Note that this check is in two places to cover API updates and Message updates. I don't love that but I wasn't finding a common spot that both flows would hit early enough before something else blew up. The duplicate usage:

https://github.com/nasa/cumulus/pull/3082/files#diff-89b5b659c17117268df6778d4734f4bc575e68360a46f3334f7b9cc4e32acde3

Reply:
given this is a private method, I don't hate the check being here.   As such, though, logically this method is just part of the message logic - we're already custom near-duplicating because of divergant expectations between message and api.     That could possibly be refactored but in the context of this PR that seems fine.